dashboards/k8s-storage-resources-glusterfs-pv.libsonnet

```diff
@@ -0,0 +1,272 @@
+          g.statPanel('')
+        )
+        .addPanel(
+          g.panel('Rebalance-Skipped') +
+          statPanelArgs +
+          g.statPanel('')
+        )
+        .addPanel(
+          g.panel('Rebalance-Failed') +
+          statPanelArgs +
+          g.statPanel('')
+        )
+        .addPanel(
+          g.panel('Capacity Utilization') +
+          graphPanelArgs +
+          g.queryPanel('gluster:volume_capacity_used_bytes_total:sum{job = "$OCS_Cluster",volume = "$Volume"}', '{{$Volume}}') +
+          g.stack + 
+          { yaxes: g.yaxes('decbytes') },
+        )
+        .addPanel(
+          g.panel('IOPS') +
+          graphPanelArgs +
+          g.queryPanel('sum(namespace_pod_name_container_name:container_cpu_usage_seconds_total:sum_rate) by (namespace)', '{{namespace}}') +
+          g.stack
+        )
+      )
+      .addRow(
+        g.row('LVM Summary')
+        .addPanel(
+          g.panel('LVM Thin Pool Metadata %') +
+          g.queryPanel('100 * gluster:volume_capacity_used_bytes_total:sum{volume = "$Volume"} / gluster:volume_capacity_available_bytes_total:sum{volume = "$Volume"}', '{{namespace}}') +
+          g.stack
+        )
+        .addPanel(
+          g.panel('LVM Thin Pool Data Usage  %') +
+          g.queryPanel('sum(namespace_pod_name_container_name:container_cpu_usage_seconds_total:sum_rate) by (namespace)', '{{namespace}}') +
+          g.stack
+        )
+      )
+      .addRow(
+        (g.row('Bricks (Volume Subcomponent) Summary') +
+         {
+           showTitle: false,
+           panels: self._panels,
+         })
+        .addPanel(
+          g.panel('Bricks-Total') +
+          brickStatPanelArgs +
+          g.statPanel('gluster_volume_brick_count{job = "$OCS_Cluster", volume = "$Volume" }','none')
+         )
+         .addPanel(
+          g.panel('Bricks-Up') +
+          brickStatPanelArgs +
+          g.statPanel('count (gluster_brick_up{job = "$OCS_Cluster", volume = "$Volume" } == 1)','none')+
+          {
+            thresholds: "0.1,0.9",
+            "colors": [
+              "#FFFFFF",
+              "#FFFFFF",
+              "#92d400",
+            ],
+            colorValue: true,
+          }
+         )
+         .addPanel(
+          g.panel('Bricks-Down') +
+          brickStatPanelArgs +
+          g.statPanel('count(gluster_brick_up{job = "$OCS_Cluster", volume = "$Volume" }) - count (gluster_brick_up{job = "$OCS_Cluster", volume = "$Volume" } == 1)','none')+
+          {
+            thresholds: "0.1,0.9",
+            "colors": [
+              "#FFFFFF",
+              "#FFFFFF",
+              "#cc0000",
+            ],
+            colorValue: true,
+          }
+         )
+        .addPanel(
+          g.panel('Capacity Utilization-Average') +
+          brickStatPanelArgs +
+          g.statPanel('avg (gluster_brick_capacity_used_bytes{job = "$OCS_Cluster",volume = "$Volume"} / gluster_brick_capacity_bytes_total{job = "$OCS_Cluster",volume = "$Volume"})')
```

Comment:
How do we know that this is the average?
Shouldn't this be calculated for a range of collected metrics?

Reply:
Since gluster_brick_capacity_used_bytes is a brick level metrics , there will me multiple metrics for a multiple bricks under one volume. It is not nacessary for each brick to have same data at certain point. That is why the AVG is being used in the query.